src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -367,6 +363,13 @@ public virtual string CustomMethod
         [Parameter]
         public virtual SwitchParameter PreserveHttpMethodOnRedirect { get; set; }
 
+        /// <summary>
+        /// Gets or sets the UnixSocket property.
+        /// </summary>
+        [Parameter]
+        [ValidateNotNullOrEmpty]
+        public virtual UnixDomainSocketEndPoint UnixSocket { get; set; }
```

Comment:
We should check `Socket.OSSupportsUnixDomainSockets` and throw if `false`.

Reply:
@xtqqczze done

Reply:
@CarloToso I don't see where this is being handled?  Presumably if the OS doesn't support UnixSockets then an appropriate message is being returned?

Reply:
@SteveL-MSFT User should create UnixDomainSocketEndPoint **before** he could pass it to the parameter.

Reply:
> We should handle the case where `Socket.OSSupportsUnixDomainSockets is false`, for example on Windows 2012/R2 and Windows 10 1607, which are still supported.

I agree with @iSazonov, error handling is not required as it should not be possible to initialize a `UnixDomainSocketEndPoint` object when `Socket.OSSupportsUnixDomainSockets` is `false`.